### PR TITLE
Print statement when trying to download empty spreadsheet and make sure sockets are closed

### PR DESCRIPTION
**Pull request by justcalamari**
Solves #31, #32 and #34.

### CODE REVIEW

File: news/empty_download.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
+**Added:** None
+
+**Changed:**
+
+* Print message when attempting to download an empty spreadsheet instead of raising an error.
+* Downloaded spreadsheets have columns in the correct order.
```
Comment (CJ-Wright) [inline news/empty_download.rst]: I think we may need a newline between these two.

File: sampledb/reader.xsh [hunk 2]
```diff
@@ -68,15 +69,19 @@ def upload_samples(host=None, db='sampleDB', collection='samples', key=None, use
...
+        if len(known):
+            print('The following sample uids were found in the database:')
+        for uid in known:
+            print(uid)
+        write_sample_spreadsheet(unknown, sdb)
+    except Exception as e:
+        print(e)
+    finally:
+        if key:
+            ssh -S sock -O exit @(server)
```
Comment (CJ-Wright) [inline sampledb/reader.xsh]: I think that `SampleDatabase` may be a good [context manager](https://www.python.org/dev/peps/pep-0343/).

In this way we can internalize the ssh tunnel/socket handling into the database itself and remove that from the rest of the code.

```python
with SampleDatabase(host, db, collection, key, user, port) as sdb:
    uids = get_uid_from_qr()
    known, unknown = find_uids(uids, sdb)
    ....
    write_sample_spreadsheet(unknown, sdb)
```

File: sampledb/sampledatabase.py [hunk 3]
```diff
@@ -7,16 +7,34 @@
 
 class SampleDatabase(object):
 
-    def __init__(self, hostname=None, db='sampleDB', collection='samples'):
-        c = MongoClient(hostname)
+    def __init__(self, hostname=None, db='sampleDB', collection='samples', key, user, port):
```
Comment (CJ-Wright) [inline sampledb/sampledatabase.py]: Can't have args after kwargs

File: sampledb/sampledatabase.py [hunk 4]
```diff
@@ -7,16 +7,34 @@
...
+        self.collection = collection
+        self.key = key
+        self.user = user
+        self.port = port
+
+    def __enter__(self):
+        if self.key:
+            server = self.user + '@' + self.hostname
+            tunnel = str(self.port) + ':localhost:27017'
+            ssh -i @(self.key) -fNMS sock -L @(tunnel) @(server)
```
Comment (CJ-Wright) [inline sampledb/sampledatabase.py]: This is currently a python file so xonsh syntax won't work.

**Comment by justcalamari** [comment]
I believe this solves #35 now:
~~~~
with SampleDatabase(host, db, collection, key, user, port) as sdb:
    sdb.publish(1520224107.xlsx)
~~~~

**Comment by CJ-Wright** [comment]
Perfect!

**Pull request merged**